In [5]:
tic

ticker
0         A
1        AA
2      AADI
3       AAL
4       AAN
...     ...
2496    MPW
2497   MPWR
2498    MPX
2499     MQ
2500   MRAM

[2501 rows x 1 columns]

In [ ]:
import requests
from bs4 import BeautifulSoup
import random
import time
import sys
import argparse
import logging
import traceback
import pandas as pd


# header generator
# Function 2
def random_user_agent():
    # https://techblog.willshouse.com/2012/01/03/most-common-user-agents/
    agents = [
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36'
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:82.0) Gecko/20100101 Firefox/82.0',
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.66 Safari/537.36',
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:83.0) Gecko/20100101 Firefox/83.0',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.1 Safari/605.1.15',
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36',
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36',
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.193 Safari/537.36',
        'Mozilla/5.0 (Windows NT 10.0; rv:78.0) Gecko/20100101 Firefox/78.0',
        'Mozilla/5.0 (X11; Linux x86_64; rv:82.0) Gecko/20100101 Firefox/82.0',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36',
        'Mozilla/5.0 (X11; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0',
        'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:82.0) Gecko/20100101 Firefox/82.0',
        'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0',
        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.111 Safari/537.36',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.67 Safari/537.36',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.1 Safari/605.1.15',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:82.0) Gecko/20100101 Firefox/82.0',
        'Mozilla/5.0 (Macintosh; Intel Mac OS X 11_0_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36'
    ]
    return random.choice(agents)


def get_headers():
    headers = {
        'authority': 'finance.yahoo.com',
        'cache-control': 'max-age=0',
        'upgrade-insecure-requests': '1',
        'user-agent': random_user_agent(),
        'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
        'sec-fetch-site': 'none',
        'sec-fetch-mode': 'navigate',
        'sec-fetch-user': '?1',
        'sec-fetch-dest': 'document',
        'accept-language': 'en-IN,en-US;q=0.9,en;q=0.8,hi;q=0.7',
    }

    return headers


# Required arrays
company_name = []
Exchange = []
market_cap = []
market_cap_raw = []
sectors = []
industrys = []
pe_ratio = []
ticker = []

# Required url
company_info_url = "https://finance.yahoo.com/quote/{stock}/profile?p={stock}"
company_stats_url = "https://finance.yahoo.com/quote/{stock}/?p={stock}"
company_history_data_url = "https://finance.yahoo.com/quote/{stock}/history?p={stock}"

tic = pd.read_excel(r'D:\Summer Internship\EY Internship\Finance project\tic_2.xlsx')
urls = []
# generating urls for the tickers
for i in range(len(tic['ticker'])):
# for i in range(2000,2501):

    url = "https://query1.finance.yahoo.com/v7/finance/quote?" \
          "formatted=true&crumb=taPfWOBv4PP&lang=en-US&" \
          "region=US&symbols=" + str(tic['ticker'][i]) + "&fields=messageBoardId%2C" \
                                                         "longName%2CshortName%2CmarketCap%2CunderlyingSymbol%2C" \
                                                         "underlyingExchangeSymbol%2CheadSymbolAsString%2C" \
                                                         "regularMarketPrice%2CregularMarketChange%2C" \
                                                         "regularMarketChangePercent%2C" \
                                                         "regularMarketVolume%2Cuuid%2C" \
                                                         "regularMarketOpen%2C" \
                                                         "fiftyTwoWeekLow%2CfiftyTwoWeekHigh%2C" \
                                                         "toCurrency%2CfromCurrency%2CtoExchange%2C" \
                                                         "fromExchange&corsDomain=finance.yahoo.com"
    urls.append(url)


# for i in range(2400,2501):
for i in range(len(urls)):

    payload = {}
    headers = {
        'authority': 'query1.finance.yahoo.com',
        'user-agent': random_user_agent(),
        'accept': '*/*',
        'origin': 'https://finance.yahoo.com',
        'sec-fetch-site': 'same-site',
        'sec-fetch-mode': 'cors',
        'sec-fetch-dest': 'empty',
        'referer': 'https://finance.yahoo.com/quote/AAPL?p=AAPL',
        'accept-language': 'en-IN,en-US;q=0.9,en;q=0.8,hi;q=0.7'
    }
    response = requests.request("GET", urls[i], headers=headers, data=payload)
    response = response.json()

    response = response.get('quoteResponse')
    result = response.get('result')
    
    if result==None or len(result)==0:
        company_name.append("N/A")
        Exchange.append("N/A")
        market_cap.append("N/A")
        market_cap_raw.append("N/A")
        sectors.append("N/A")
        industrys.append("N/A")
        pe_ratio.append("N/A")
        ticker.append(tic['ticker'][i])
        
    elif len(result)>0 and result!=None:

        try:    
            ticker.append(tic['ticker'][i])
            if len(result) > 0 and result != None:
                result = result[0]
                default_d = {'fmt': '', 'raw': ''}
                if 'longName' in result:
                    if len(result.get('longName').replace(",", "")) > 0:

                        company_name.append(result.get('longName').replace(",", ""))
                    else:
                        company_name.append("N/A")
                else:
                    url = company_info_url.format(stock=tic["ticker"][i])
                    response = requests.request("GET", url, headers=get_headers(), data={})
                    # print(response.text)
                    soup = BeautifulSoup(response.text, 'html.parser')
                    a = soup.find_all('h1', {'class': "D(ib) Fz(18px)"})
                    if len(a) > 0:
                        company_name.append(soup.find_all('h1', {'class': "D(ib) Fz(18px)"})[0].text)
                    else:
                        company_name.append("N/A")

                if len(str(result.get('fullExchangeName'))) > 0:
    #                 if str(result.get('fullExchangeName'))="NYSE" or rstr(result.get('fullExchangeName'))="NYSE"
                    Exchange.append((str(result.get('fullExchangeName'))))
                else:
                    Exchange.append("N/A")
                if len(str(result.get('marketCap', default_d).get('fmt'))) > 0:
                    market_cap.append(str(result.get('marketCap', default_d).get('fmt')))
                else:
                    market_cap.append("N/A")
                if len(str(result.get('marketCap', default_d).get('raw'))) > 0:
                    market_cap_raw.append(str(result.get('marketCap', default_d).get('raw')))
                else:
                    market_cap_raw.append("N/A")

            else:
                company_name.append("N/A")
                Exchange.append("N/A")
                market_cap.append("N/A")
                market_cap_raw.append("N/A")

            # Sectors and Industry

            try:
                url = company_info_url.format(stock=tic["ticker"][i])
                response = requests.request("GET", url, headers=get_headers(), data={})
                # print(response.text)
                soup = BeautifulSoup(response.text, 'html.parser')
                # print(soup)

                sector_industry = soup.find_all('span', {'class': 'Fw(600)'})
                sector = sector_industry[0].text
                sector = sector.replace(',', '')
                industry = sector_industry[1].text
                industry = industry.replace(',', '')
                sectors.append(sector)
                industrys.append(industry)

            except:
                sectors.append("N/A")
                industrys.append("N/A")

            # pe Ratio
            try:

                url = company_stats_url.format(stock=tic["ticker"][i])
                response = requests.request("GET", url, headers=get_headers(), data={})
                soup = BeautifulSoup(response.text, 'html.parser')

                per = soup.find_all('td', {'data-test': 'PE_RATIO-value'})
                if len(per[0].text) > 0:
                    pe_ratio.append(float(per[0].text))
                else:
                    pe_ratio.append(0.0)
            except:
                pe_ratio.append(0.0)

            print(str(i+1)+" "+" Tickers Done"+" "+str(tic["ticker"][i]))
        except:
            company_name.append("N/A")
            Exchange.append("N/A")
            market_cap.append("N/A")
            market_cap_raw.append("N/A")
            Exchange.append("N/A")
            pe_ratio.append("N/A")
            industrys.append("N/A")
            sectors.append("N/A")
# creating data set
data1 = {
    "ticker": ticker,
    "Company_Name": company_name,
    "Sector": sectors,
    "Industry": industrys,
    "Exchange": Exchange,
    "Market_Cap_B": market_cap_raw,
    "Market_Cap": market_cap,
    "PE_Ratio": pe_ratio}

dfd = pd.DataFrame(data1)
var = len(dfd.axes[0])
try:
    for i in range(var):
        if dfd["Company_Name"][i] == 'N/A':
            dfd["Sector"][i] = 'N/A'
            dfd["Industry"][i] = 'N/A'
            dfd["Exchange"][i] = 'N/A'
            dfd["Market_Cap_B"][i] = 'N/A'
            dfd["Market_Cap"][i] = 'N/A'
            dfd["PE_Ratio"][i] = 'N/A'
except:
    print("Contact the Developer")

dfd.to_csv("Yahoo_finance_ticker_data_updated_3.csv", index=False)
print("Done! Please open file Yahoo_finance_ticker_data_3.csv")


1  Tickers Done DIGP
2  Tickers Done DIIBF
3  Tickers Done DIN
4  Tickers Done DINRF
5  Tickers Done DIOD
6  Tickers Done DIS
10  Tickers Done DISH
11  Tickers Done DIT
12  Tickers Done DJCO
13  Tickers Done DK
14  Tickers Done DKDCA
15  Tickers Done DKILY
16  Tickers Done DKL
17  Tickers Done DKNG
18  Tickers Done DKS
19  Tickers Done DLA
20  Tickers Done DLAKY
21  Tickers Done DLB
22  Tickers Done DLEXY
23  Tickers Done DLHC
24  Tickers Done DLNG
25  Tickers Done DLO
26  Tickers Done DLPN
27  Tickers Done DLR
28  Tickers Done DLTH
29  Tickers Done DLTR
30  Tickers Done DLX
31  Tickers Done DM
32  Tickers Done DMAC
33  Tickers Done DMIFF
34  Tickers Done DMLP
35  Tickers Done DMRC
36  Tickers Done DMS
37  Tickers Done DMTK
38  Tickers Done DNA
39  Tickers Done DNACF
40  Tickers Done DNAY
41  Tickers Done DNB
42  Tickers Done DNBBY
43  Tickers Done DNFGY
44  Tickers Done DNKEY
45  Tickers Done DNLI
46  Tickers Done DNMR
47  Tickers Done DNN
48  Tickers Done DNOW
49  Tickers Done DNPLY


375  Tickers Done EXN
376  Tickers Done EXP
377  Tickers Done EXPD
378  Tickers Done EXPE
379  Tickers Done EXPI
380  Tickers Done EXPO
381  Tickers Done EXPR
382  Tickers Done EXR
383  Tickers Done EXTN
384  Tickers Done EXTR
385  Tickers Done EYE
386  Tickers Done EYEN
387  Tickers Done EYPT
388  Tickers Done EZFL
389  Tickers Done EZPW
390  Tickers Done F
391  Tickers Done FA
392  Tickers Done FACO
393  Tickers Done FAF
394  Tickers Done FALC
395  Tickers Done FANG
396  Tickers Done FANH
397  Tickers Done FANUY
398  Tickers Done FARM
399  Tickers Done FARO
400  Tickers Done FAST
401  Tickers Done FAT
402  Tickers Done FATBB
403  Tickers Done FATE
404  Tickers Done FB
405  Tickers Done FBAK
406  Tickers Done FBC
407  Tickers Done FBHS
408  Tickers Done FBIO
409  Tickers Done FBIZ
410  Tickers Done FBK
411  Tickers Done FBMS
412  Tickers Done FBNC
413  Tickers Done FBP
414  Tickers Done FBRT
415  Tickers Done FBRX
416  Tickers Done FBSI
417  Tickers Done FC
418  Tickers Done FCAP
419 

745  Tickers Done GL
746  Tickers Done GLAD
747  Tickers Done GLAE
748  Tickers Done GLAPY
749  Tickers Done GLBS
750  Tickers Done GLBZ
751  Tickers Done GLDD
752  Tickers Done GLDG
753  Tickers Done GLG
754  Tickers Done GLGI
755  Tickers Done GLGLF
756  Tickers Done GLMD
757  Tickers Done GLNCY
758  Tickers Done GLNG
759  Tickers Done GLOB
760  Tickers Done GLOP
761  Tickers Done GLP
762  Tickers Done GLPEY
763  Tickers Done GLPG
764  Tickers Done GLPI
765  Tickers Done GLRE
766  Tickers Done GLT
767  Tickers Done GLTO
768  Tickers Done GLUC
769  Tickers Done GLUE
770  Tickers Done GLW
771  Tickers Done GLXZ
772  Tickers Done GLYC
773  Tickers Done GM
774  Tickers Done GMAB
775  Tickers Done GMBL
776  Tickers Done GMDA
777  Tickers Done GME
778  Tickers Done GMED
779  Tickers Done GMELY
780  Tickers Done GMHLY
781  Tickers Done GMRE
782  Tickers Done GMS
783  Tickers Done GMTX
784  Tickers Done GNAL
785  Tickers Done GNBT
786  Tickers Done GNCA
787  Tickers Done GNE
788  Tickers Don

1112  Tickers Done HURC
1113  Tickers Done HURN
1114  Tickers Done HUSA
1115  Tickers Done HUSN
1116  Tickers Done HUT
1117  Tickers Done HUYA
1118  Tickers Done HVBC
1119  Tickers Done HVRRY
1120  Tickers Done HVT
1122  Tickers Done HWBK
1123  Tickers Done HWC
1124  Tickers Done HWDJY
1126  Tickers Done HWKN
1127  Tickers Done HWM
1128  Tickers Done HWNI
1129  Tickers Done HXGBY
1130  Tickers Done HXL
1131  Tickers Done HY
1132  Tickers Done HYFM
1133  Tickers Done HYLN
1134  Tickers Done HYMC
1135  Tickers Done HYPMY
1136  Tickers Done HYRE
1137  Tickers Done HYSNY
1138  Tickers Done HYZN
1139  Tickers Done HZN
1140  Tickers Done HZNP
1141  Tickers Done HZO
1142  Tickers Done IAA
1143  Tickers Done IAALF
1144  Tickers Done IAC
1145  Tickers Done IACC
1146  Tickers Done IAG
1147  Tickers Done IART
1148  Tickers Done IAS
1149  Tickers Done IBA
1150  Tickers Done IBAL
1151  Tickers Done IBCP
1152  Tickers Done IBDRY
1153  Tickers Done IBEX
1154  Tickers Done IBIO
1155  Tickers Done IBKR

1464  Tickers Done JVA
1467  Tickers Done JWEL
1468  Tickers Done JWN
1469  Tickers Done JXHLY
1470  Tickers Done JXN
1471  Tickers Done JYNT
1472  Tickers Done K
1473  Tickers Done KAI
1474  Tickers Done KAJMY
1475  Tickers Done KALA
1476  Tickers Done KALO
1477  Tickers Done KALU
1478  Tickers Done KALV
1479  Tickers Done KAMN
1480  Tickers Done KAR
1481  Tickers Done KAYS
1482  Tickers Done KB
1483  Tickers Done KBAL
1484  Tickers Done KBCSY
1485  Tickers Done KBH
1486  Tickers Done KBNT
1487  Tickers Done KBR
1488  Tickers Done KC
1489  Tickers Done KCDMY
1490  Tickers Done KCLI
1491  Tickers Done KDDIY
1492  Tickers Done KDNY
1493  Tickers Done KDOZF
1494  Tickers Done KDP
1495  Tickers Done KE
1496  Tickers Done KELYA
1497  Tickers Done KELYB
1498  Tickers Done KEN
1499  Tickers Done KEP
1500  Tickers Done KEQU
1501  Tickers Done KERN
1502  Tickers Done KEX
1503  Tickers Done KEY
1504  Tickers Done KEYS
1505  Tickers Done KFFB
1506  Tickers Done KFRC
1507  Tickers Done KFS
1508  

1821  Tickers Done LVO
1822  Tickers Done LVOX
1823  Tickers Done LVRA
1824  Tickers Done LVS
1825  Tickers Done LVTX
1826  Tickers Done LVWD
1827  Tickers Done LW
1828  Tickers Done LWAY
1829  Tickers Done LWLG
1830  Tickers Done LX
1831  Tickers Done LXEH
1832  Tickers Done LXFR
1833  Tickers Done LXP
1834  Tickers Done LXRX
1835  Tickers Done LXU
1836  Tickers Done LYB
1837  Tickers Done LYEL
1838  Tickers Done LYFT
1839  Tickers Done LYG
1840  Tickers Done LYLT
1841  Tickers Done LYRA
1842  Tickers Done LYTS
1843  Tickers Done LYV
1844  Tickers Done LZ
1845  Tickers Done LZAGY
1846  Tickers Done LZB
1847  Tickers Done M
1849  Tickers Done MA
1850  Tickers Done MAA
1851  Tickers Done MAAL
1852  Tickers Done MAC
1853  Tickers Done MACE
1854  Tickers Done MACK
1855  Tickers Done MAG
1856  Tickers Done MAIN
1857  Tickers Done MAN
1858  Tickers Done MANH
1859  Tickers Done MANT
1860  Tickers Done MANU
1861  Tickers Done MAPS
1862  Tickers Done MAR
1863  Tickers Done MARA
1864  Tickers D

In [19]:
tic['ticker'][2400]

'MESA'

In [ ]:
lt

In [ ]:
data1 = {
    "ticker": ticker,
    "Company_Name": company_name,
    "Sector": sectors,
    "Industry": industrys,
    "Exchange": Exchange,
    "Market_Cap_B": market_cap_raw,
    "Market_Cap": market_cap,
    "PE_Ratio": pe_ratio}
dfd = pd.DataFrame(data1)
dfd.to_csv("Yahoo_finance_ticker_data_updated_3.csv", index=False)

In [10]:
print('ticker', len(data['ticker']))
print('Company_Name', len(data['Company_Name']))
print('Sector', len(data['Sector']))
print('Industry', len(data['Industry']))
print('Exchange', len(data['Exchange']))
print('Market_Cap_B', len(data['Market_Cap_B']))
print('Market_Cap', len(data['Market_Cap']))
print('PE_Ratio', len(data['PE_Ratio']))

ticker 16
Company_Name 10
Sector 10
Industry 10
Exchange 10
Market_Cap_B 10
Market_Cap 10
PE_Ratio 10


In [13]:
data['Exchange']

['N/A', 'N/A', 'NYSE', 'N/A', 'None', 'N/A', 'N/A', 'None', 'N/A', 'NYSE']

In [3]:
new_data = {
    "ticker": ticker,
    "Company_Name": company_name,
    "Sector": sectors,
    "Industry": industrys,
    "Exchange": ,
    "Market_Cap_B": market_cap_raw,
    "Market_Cap": market_cap,
    "PE_Ratio": pe_ratio}


SyntaxError: invalid syntax (2721124591.py, line 6)

In [4]:
new_data2 = {"Exchange": Exchange}

In [5]:
new_data2

{'Exchange': ['NYSE',
  'NYSE',
  'NasdaqCM',
  'NasdaqGS',
  'NYSE',
  'NasdaqGM',
  'NasdaqGS',
  'NYSE',
  'NasdaqGS',
  'NYSE',
  'NasdaqCM',
  'NasdaqGS',
  'NYSE',
  'NYSE',
  'NYSE',
  'NYSE',
  'NasdaqGS',
  'NasdaqGS',
  'NasdaqGS',
  'NYSE',
  'NYSE',
  'NasdaqGS',
  'NasdaqGS',
  'NasdaqGS',
  'NYSE',
  'NasdaqGS',
  'NasdaqGS',
  'NYSE',
  'NasdaqGM',
  'NasdaqCM',
  'NYSE',
  'NYSE',
  'NasdaqGS',
  'NasdaqGS',
  'NasdaqGS',
  'NYSE',
  'NasdaqGS',
  'NYSE',
  'NasdaqGS',
  'NYSE',
  'NasdaqGS',
  'NasdaqGS',
  'NYSE',
  'NasdaqGM',
  'NasdaqGS',
  'NasdaqGS',
  'NYSE',
  'NasdaqGM',
  'NYSE',
  'NasdaqCM',
  'NYSE',
  'NasdaqGS',
  'NYSE American',
  'NasdaqGS',
  'NasdaqCM',
  'NasdaqGM',
  'NasdaqGS',
  'NasdaqGS',
  'NYSE',
  'NYSE',
  'NasdaqGM',
  'NasdaqGM',
  'NasdaqGS',
  'NYSE',
  'None',
  'NasdaqCM',
  'NYSE',
  'NasdaqGS',
  'NasdaqGS',
  'N/A',
  'NasdaqGS',
  'NYSE',
  'NasdaqGS',
  'NasdaqGS',
  'NasdaqGS',
  'NasdaqGM',
  'NasdaqCM',
  'NYSE American',
  '

In [6]:
import numpy as np
unique_exchange = set(data['Exchange'])
cd=[]
for i in unique_exchange:
    cd.append(str(i)+" "+str(data['Exchange'].count(i)))

In [7]:
cd

['NasdaqGM 247',
 'NYSE American 35',
 'NasdaqGS 851',
 'N/A 61',
 'None 36',
 'Other OTC 8',
 'NYSE 1021',
 'NYSEArca 1',
 'NasdaqCM 247']

In [49]:
data['Exchange'].count('None')

35

In [50]:
data['Exchange'].index('None')

64

In [8]:
indices = [i for i, x in enumerate(data['Exchange']) if x == "NYSEArca"]
indices

[1134]

['NasdaqGM 247',
 'NYSE American 35',
 'NasdaqGS 851',
 'N/A 61',
 'None 36',
 'Other OTC 8',
 'NYSE 1021',
 'NYSEArca 1',
 'NasdaqCM 247']

In [30]:
indices = [i for i, x in enumerate(data['Exchange']) if x.count(" ") > 3]
indices

[]

In [85]:
new_df

Empty DataFrame
Columns: [ticker, Company_Name, Sector, Industry, Exchange, Market_Cap_B, Market_Cap, PE_Ratio]
Index: []

In [ ]:
list1 = []
for i in indices:
    data1 = {
        "ticker": ticker[i],
        "Company_Name": company_name[i],
        "Sector": sectors[i],
        "Industry": industrys[i],
        "Exchange": Exchange[i],
        "Market_Cap_B": market_cap_raw[i],
        "Market_Cap": market_cap[i],
        "PE_Ratio": pe_ratio[i]}

In [88]:
data1

{'ticker': 'DIDI',
 'Company_Name': 'DiDi Global Inc.',
 'Sector': 'Technology',
 'Industry': 'Software—Application',
 'Exchange': 'NYSEArca',
 'Market_Cap_B': '11045220352',
 'Market_Cap': '11.045B',
 'PE_Ratio': 0.0}

In [ ]:
dfd

In [ ]:
lt=[]
for i in data:
    lt.append(i)

    

In [3]:
print((len(data.values())))

7


In [5]:
ft=[company_name,
Exchange,
market_cap,
market_cap_raw,
sectors,
industrys,
pe_ratio,
ticker]

In [6]:
for i in ft :
    print(" ",print(len(i)))

2501
  None
2508
  None
2501
  None
2501
  None
2501
  None
2501
  None
2501
  None
2501
  None


In [ ]:
https://query1.finance.yahoo.com/v7/finance/quote?\formatted=true&crumb=taPfWOBv4PP&lang=en-US&\region=US&symbols=A&fields=messageBoardId%2C\longName%2CshortName%2CmarketCap%2CunderlyingSymbol%2C\underlyingExchangeSymbol%2CheadSymbolAsString%2C\regularMarketPrice%2CregularMarketChange%2C\regularMarketChangePercent%2C\regularMarketVolume%2Cuuid%2C\regularMarketOpen%2C\fiftyTwoWeekLow%2CfiftyTwoWeekHigh%2C\toCurrency%2CfromCurrency%2CtoExchange%2C\fromExchange&corsDomain=finance.yahoo.com

In [6]:
for i in range(246,345):
    print(i)

246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
277
278
279
280
281
282
283
284
285
286
287
288
289
290
291
292
293
294
295
296
297
298
299
300
301
302
303
304
305
306
307
308
309
310
311
312
313
314
315
316
317
318
319
320
321
322
323
324
325
326
327
328
329
330
331
332
333
334
335
336
337
338
339
340
341
342
343
344
